pkg/server/server_controller.go

```diff
@@ -110,11 +110,12 @@ type serverController struct {
 		// TODO(knz): Detect when the mapping of name to tenant ID has
 		// changed, and invalidate the entry.
 		servers map[roachpb.TenantName]*serverEntry
-
-		// nextServerIdx is the index to provide to the next call to
-		// newServerFn.
-		nextServerIdx int
 	}
+
+	// nextServerIdx is the index to provide to the next call to
+	// newServerFn. This is updated concurrently, use the atomic
+	// package to access.
+	nextServerIdx uint32
```

Comment:
How do you feel about the atomic types: https://pkg.go.dev/sync/atomic#Uint32

They were added in go 1.19. A couple of nice properties is that they trigger the NoCopy go vet rule and for 64-bit types (not relevant here) you don't need to worry about alignment.